tests/cases/conformance/jsdoc/paramTagOnFunctionUsingArguments.ts

```diff
@@ -0,0 +1,18 @@
+// @noEmit: true
+// @allowJs: true
+// @checkJs: true
+// @strict: true
+// @Filename: decls.d.ts
+declare function factory(type: string): {};
+// @Filename: a.js
+
+/**
+ * @param {string} first
```

Comment:
This case I'm not so sure about supporting since people will think they're getting typing, when we just throw the type definition away without telling them and accept `any`. Instead we could add a better message telling them to use a rest parameter instead.

Reply:
Rest parameters don't exist in ES5, so the error would only apply to ES2015 and later. And we don't usually know whether a .js file targets ES2015 or later. 

As for throwing away typing, I don't think there's much we can do -- maybe a noImplicitAny error instead?

I think it would be easier to error (to info-error?) whenever there's a reference to `arguments`. That error would be a lot easier to write, and would apply to both ES2015+ and TS, and should come with an associated quick fix. I'll open a separate issue for that.

Reply:
Currently we require the user to place a dummy-parameter and type it as a rest parameter:
```ts
/** @param {...string} x */
function f(x) {
    return arguments[0];
}
```
We could make a codefix to convert `@param {string} first` to `@param {...string} args` and add the dummy parameter -- we could also stop requiring the dumy parameter.

Reply:
I like that. I'll see how hard it is to get the error for the `{string} x` case and then remove it when it's changed to `{...string} x`. We can do the code fix later, since @mhegazy wants to get this into 2.8.1.